core/dbt/task/base.py

```diff
@@ -45,10 +45,11 @@
 from dbt.events.contextvars import get_node_info
 from .printer import print_run_result_error
 
-from dbt.adapters.factory import register_adapter
+from dbt.adapters.factory import get_adapter, register_adapter
 from dbt.config import RuntimeConfig, Project
 from dbt.config.profile import read_profile
 import dbt.exceptions
+from dbt.graph import Graph
 
 
 class NoneConfig:
```

Comment:
Do we plan to keep this in longer term? Or just something to help during the transition period?

Reply:
Compilation needed to live somewhere, and since the ask of this ticket is only abstracting manifest generation, not compilation, I moved the behavior to `ConfiguredTask`. If we want to move this behavior elsewhere, we can do so in the future.